### PR TITLE
Prepare jobs for release 0.9

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Prepare jobs for release 0.9
- Define branch protector rules
- Change release version to 0.9.0-rc1 
- Remove jobs for 0.6 release

**Related issue(s)**
See https://github.com/kyma-project/kyma/issues/3330

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -325,6 +325,14 @@ branch-protection:
                 - pre-rel08-kyma-gke-integration
                 - pre-rel08-kyma-artifacts
                 - pre-rel08-kyma-installer
+            release-0.9:
+              protect: true
+              required_status_checks:
+                contexts:
+                  - pre-rel09-kyma-integration
```
Comment (aszecowka) [inline prow/config.yaml]: probably upgrade job should also be required

File: prow/jobs/kyma/kyma-integration.yaml [hunk 2]
```diff
@@ -193,6 +205,20 @@ presubmits: # runs on PRs
       path_alias: github.com/kyma-project/test-infra
       base_ref: release-0.8
 
+  - name: pre-rel09-kyma-gke-integration
```
Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: when are we going to remove jobs for older releases?

File: prow/config.yaml [hunk 3]
```diff
@@ -325,6 +325,14 @@ branch-protection:
...
+                  - pre-rel09-kyma-gke-integration
```
Comment (aszecowka) [inline prow/config.yaml]: what about central connector job?

File: prow/jobs/kyma/kyma-integration.yaml (hunk 2 above)
Comment (pkosiec) [reply prow/jobs/kyma/kyma-integration.yaml]: Good question. I confirmed with @jose-cortina that we need only 3 jobs. That's why I removed release-0.6 jobs in this pull request.

**Pull request merged**